DylibBinder/DBFunc.cs

```diff
@@ -0,0 +1,115 @@
+			IsProperty = true;
+			HasThrows = false;
+			IsMutating = isMutating;
+			HasInstance = !IsStatic;
+			ReturnType = dbProperty.Type;
+			ParameterLists = new DBParameterLists (HasInstance, propertyType);
+			GenericParameters = dbProperty.GenericParameters;
+		}
+
+		public string Name { get; }
+		public bool IsStatic { get; }
+		public bool IsProperty { get; }
+		public string ReturnType { get; }
+		public bool HasThrows { get; }
+		public bool IsMutating { get; }
+		public bool HasInstance { get; }
+
+		public string PropertyType { get; } = string.Empty;
+		public bool IsPossiblyIncomplete { get; } = false;
+		public OperatorType OperatorKind { get; } = OperatorType.None;
+		public TypeAccessibility Accessibility { get; } = TypeAccessibility.Public;
+		public bool IsFinal { get; } = false;
+		public bool IsDeprecated { get; } = false;
+		public bool IsUnavailable { get; } = false;
+		public bool IsOptional { get; } = false;
+		public bool IsRequired { get; } = false;
+		public bool IsConvenienceInit { get; } = false;
+		public string ObjcSelector { get; } = string.Empty;
+
+		public DBParameterLists ParameterLists { get; }
+		public DBGenericParameters GenericParameters { get; }
+		public DBAssociatedTypes AssociatedTypes { get; } = new DBAssociatedTypes ();
+
+		bool GetStaticStatus (SwiftBaseFunctionType signature) => signature switch {
+			SwiftStaticFunctionType => true,
+			SwiftPropertyType propType => propType.IsStatic,
+			_ => false,
+		};
+	}
+
+	internal class DBFuncs : IAssociatedTypes {
+		public DBFuncs (ClassContents classContents)
+		{
+			var functions = SortedSetExtensions.Create<OverloadInventory> ();
+			functions.AddRange (classContents.Methods.Values, classContents.StaticFunctions.Values, classContents.Constructors.Values);
+
+			foreach (var function in functions) {
+				foreach (var overloadFunction in function.Functions) {
+					if (overloadFunction.Name.Name.IsPublic () && !IsMetaClass (overloadFunction.Signature))
+						Funcs.Add (new DBFunc (overloadFunction));
+				}
+			}
+			AssociatedTypes.Add (Funcs.GetChildrenAssociatedTypes ());
+		}
+
+		public DBFuncs (ProtocolContents protocolContents)
+		{
+			var functions = SortedSetExtensions.Create<TLFunction> ();
+			functions.AddRange (protocolContents.FunctionsOfUnknownDestination);
+			foreach (var function in functions) {
+				if (function.Name.Name.IsPublic () && !IsMetaClass (function.Signature))
+					Funcs.Add (new DBFunc (function, isProtocol: true));
+			}
+			AssociatedTypes.Add (Funcs.GetChildrenAssociatedTypes ());
+		}
+
+		public List<DBFunc> Funcs { get; } = new List<DBFunc> ();
+		public DBAssociatedTypes AssociatedTypes { get; } = new DBAssociatedTypes ();
+
+		bool IsMetaClass (SwiftBaseFunctionType funcType)
+		{
+			if (funcType.ReturnType.Type == CoreCompoundType.MetaClass)
+				return true;
+			foreach (var param in funcType.EachParameter) {
+				if (param.Type == CoreCompoundType.MetaClass)
+					return true;
+			}
+			return false;
```

Comment:
```suggestion
                        return funcType.ReturnType.Type == CoreCompoundType.MetaClass
                            || funcType.EachParameter.Any(p =>  p.Type == CoreCompoundType.MetaClass);
```

Reply:
that return statement removes the entire method body, even more:
```suggestion
	bool IsMetaClass (SwiftBaseFunctionType funcType)
	    => return funcType.ReturnType.Type == CoreCompoundType.MetaClass
                            || funcType.EachParameter.Any(p =>  p.Type == CoreCompoundType.MetaClass);
```

Reply:
Ah much better!